relayer/listener.go

```diff
@@ -194,14 +248,44 @@ func (lstnr *Listener) calculateStartingBlockHeight(processHistoricalBlocksFromH
 
 func (lstnr *Listener) setAllProcessedBlockHeightsToLatest() error {
 	for _, relayer := range lstnr.applicationRelayers {
-		_, err := relayer.setProcessedBlockHeightToLatest()
+		height, err := lstnr.getCurrentHeight()
+		if err != nil {
+			return err
+		}
+		relayer.dbManager.PrepareHeight(relayer.relayerID, height, 0)
```

Comment:
`keyManager.prepareHeight` immediately commits the height if the number of messages to process is `0`. I don't think we should directly expose `commitHeight` to `dbManager` callers. I can add separate method with a clearer name that makes this call internally though.

Reply:
So with the min-heap implemented, `prepareheight` with 0 messages should only write the height if the current height is N and we commit N+1. We need a separate helper to force-set the height to latest here

Reply:
`commitHeight` automatically commits if `km.maxCommittedHeight == 0`, which is the case on initialization, and should be the case when this function is called, since it is called before the main processing loop in `ProcessLogs` is initiated.

I agree that this is not at all clear though. I'll add a more specific method that can be called here, and update `commitHeight` to accept a "force" option.